CMakeLists.txt

```diff
@@ -75,6 +75,39 @@ message(STATUS "QGIS version: ${COMPLETE_VERSION} ${RELEASE_NAME} (${QGIS_VERSIO
 
 set (ENABLE_LOCAL_BUILD_SHORTCUTS FALSE CACHE BOOL "Disables some build steps which are only relevant for releases to speed up compilation time for development")
 
+# Add the win32 resource for the QGIS icon
+# Only runs for WIN32 when called
+function(win32_icon SRC_LIST)
+  if(NOT WIN32)
+    return()
+  endif()
+
+  set(OUT "${PROJECT_SOURCE_DIR}/ms-windows/rc/icon.rc")
+  list(APPEND ${SRC_LIST} "${OUT}")
+  message(DEBUG "Win32 Icon Resource")
+
+  set(${SRC_LIST} "${${SRC_LIST}}" PARENT_SCOPE)
+endfunction()
+
+# Function to create win32 VERSIONINFO and resource blocks
+# Only runs for WIN32 when called
+function(win32_version_info desc filename SRC_LIST)
```

Comment:
I guess we can remove this (and the ones in CreateQgsVersion are kept)?

Reply:
My main reason for keeping them out of the CREATE_QGSVERSION is because they are slightly different and I didn't want to risk breaking anything inside that macro

Reply:
You also added a function with the same name in cmake/CreateQgsVersion.cmake (see https://github.com/qgis/QGIS/pull/62516/files#diff-ee874e32953e5696a8a87151cc69c723f92903edd0cc22a6cbfe20dc4978b744R78)

I would like to keep this out of the same CMakeLists.txt file, that file is huge already.

Reply:
Oh yeah oops I was meant to remove that other one in the main cmake file
after I moved it. Sorry I misread the one you highlighted and messed up my commits.